src/Microsoft.ComponentDetection.Contracts/Mappers/CycloneDx.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using CycloneDX.Json;
+using CycloneDX.Models;
+using Microsoft.ComponentDetection.Contracts.BcdeModels;
+
+namespace Microsoft.ComponentDetection.Contracts.Mappers
+{
+    public static class CycloneDx
+    {
+        public static string ToCycloneDx(this ScanResult scanResult)
+        {
+            return Serializer.Serialize(
+                new Bom
+                {
+                    Metadata = new Metadata
```

Comment:
It would also be good to set the BOM serial number. This is used to uniquely identify this CycloneDX BOM. And is particularly handy for situations like referencing BOMs of components in larger complex systems. Or for publishing to and retrieving from a BOM repository.

Using the `CycloneDX.Utils` namespace and `SerialNumber = CycloneDXUtils.GenerateSerialNumber()` should do the trick.

Reply:
There's a similar field in SPDX documents too. Doco here for when you get to it https://spdx.github.io/spdx-spec/document-creation-information/#63-spdx-identifier-field

Reply:
Thanks! Done for CycloneDX